File: CookMaster/Models/RecipeSearchResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CookMaster.Models
{
    public class RecipeSearchResult
    {
        public int ID { get; set; }
        public string Image { get; set; }
        public string ImageType { get; set; }
        public int Likes { get; set; }

        public int MissedIngredientCount { get; set; }
        public List<IngredientSearchItem> MissedIngredients { get; set; }

        public string Title { get; set; }

        public List<IngredientSearchItem> UnusedIngredients { get; set; }

        public int UsedIngredientCount { get; set; }
        public List<IngredientSearchItem> UsedIngredients { get; set; }
    }
}

File: CookMaster.WebApp/ClientApp/src/app/shared/models/recipe-search-result.ts
import {IngredientSearchItem} from './ingredient-search-item';

export interface RecipeSearchResult {
  ID: number;
  Image: string;
  ImageType: string;
  Likes: number;
  MissedIngredientCount: number;
  MissedIngredients: IngredientSearchItem[];
  Title: string;
  UnusedIngredients: IngredientSearchItem[];
  UsedIngredientCount: number;
  UsedIngredients: IngredientSearchItem[];
}
